ansible/roles/provision-vm/tasks/redhat.yml

```diff
@@ -1,13 +1,38 @@
 ---
-- name: Set distro
-  set_fact:
-    distro: "centos"
+- set_fact:
+    distro: centos
+
+- name: Get distro info
+  shell: awk -F= '$1=="ID" { print $2 ;}' /etc/os-release
+  register: distro_id
+  when: arch_info == "s390x"
+
+- set_fact:
+    distro: "{{ distro_id.stdout }}"
+  when: arch_info == "s390x"
+
+- name: Register server with Red Hat for IBM Z vms
+  tags: attach_subscription
+  community.general.redhat_subscription:
+    state: present
+    username: "{{ redhat_username }}"
+    password: "{{ redhat_password }}"
+    auto_attach: true
+  register: registration
+  retries: 2
+  delay: 30
+  until: registration is not failed
+  when: arch_info == "s390x"
+
+- debug:
+    var: distro
```

Comment:
Do we need to keep this debug statement? Doesn't hurt, I know, but still.

Reply:
good point, I'll remove